### PR TITLE
Name sandboxes archive after Jenkins job.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
Currently the sandboxes archive is called `sandboxes.tar.gz`. To make
debugging a little easier this patch will rename it to `ci-<Jenkins job>-sandboxes.tar.gz`.

### CODE REVIEW

**Comment by zen-dog** [comment]
File names

Since we're changing the names - could we please simplify it to `ci-PR-NNNN-1.*?

File: ci/pipeline [hunk 1]
```diff
@@ -249,7 +251,13 @@ def pr(): Unit = asPullRequest {
 @main
 def loop(): Unit = {
   provisionHost()
-  build()
+
+  val loopNamePattern = """marathon-sandbox/(.*)""".r
```
Comment (zen-dog) [inline ci/pipeline]: small helper method?

File: ci/pipeline [hunk 2]
```diff
@@ -130,8 +131,7 @@ def createTarballPackages(): SemVer = utils.stage("Package Tarballs") {
  * The pipeline target for GitHub pull request builds. It wraps other targets
  * and does some additional reporting to GitHub.
  */
-def asPullRequest(run: => (SemVer, Option[awsClient.Artifact])): Unit = {
-  val pullNumber: String = sys.env.getOrElse("CHANGE_ID", throw new IllegalArgumentException("No CHANGE_ID was defined."))
+def asPullRequest(pullNumber: String)(run: => (SemVer, Option[awsClient.Artifact])): Unit = {
```
[the remark this answers is not in the document]
Comment (jeschkies) [reply ci/pipeline]: It's passed on to the github client.

File: ci/utils.sc [hunk 3]
```diff
@@ -113,11 +113,11 @@ def isMasterBuild(): Boolean = {
 }
 
 /**
- * @return True if build is for pull request.
+ * @return Pull request number if build is for pull request or None if not.
  */
-def isPullRequest(): Boolean = {
+def isPullRequest(): Option[String] = {
```
[the remark this answers is not in the document]
Comment (jeschkies) [reply ci/utils.sc]: Done. ✅

File: ci/pipeline [hunk 4]
```diff
@@ -84,10 +84,11 @@ def checkSystemIntegrationTests(logFileName: String): Unit = {
  * Compresses sandboxes and logs.
  *
  * @param logFileName Name of log file.
+ * @param sandboxFilePrefix The start of the file name for the sandboxes archive.
  */
 @main
-def zipLogs(logFileName: String = "ci.log"): Unit = {
-  Try(%("tar", "--exclude=provisioner", "-zcf", "sandboxes.tar.gz", "sandboxes"))
+def zipLogs(logFileName: String = "ci.log", sandboxFilePrefix: String = "sandboxes"): Unit = {
+  Try(%("tar", "--exclude=provisioner", "-zcf", s"$sandboxFilePrefix.tar.gz", "sandboxes"))
```
[the remark this answers is not in the document]
Comment (jeschkies) [reply ci/pipeline]: This is a confusing part. The log file that is compressed has a proper name. The sandboxes folder is just called `sandboxes` but the archive should get a different name. I dropped the `Prefix`.

File: ci/pipeline [hunk 5]
```diff
@@ -219,7 +220,7 @@ def updateDcosImage(version: SemVer, artifactUrl: String, sha1: String): Unit =
 def master(): Unit = {
   provisionHost()
 
-  val version = build()
+  val version = build(buildName = "master")
```
[the remark this answers is not in the document]
Comment (jeschkies) [reply ci/pipeline]: Yes. ✅

File: ci/pipeline [hunk 6]
```diff
@@ -84,11 +84,11 @@ def checkSystemIntegrationTests(logFileName: String): Unit = {
  * Compresses sandboxes and logs.
  *
  * @param logFileName Name of log file.
- * @param sandboxFilePrefix The start of the file name for the sandboxes archive.
+ * @param sandboxFile The start of the file name for the sandboxes archive.
  */
 @main
-def zipLogs(logFileName: String = "ci.log", sandboxFilePrefix: String = "sandboxes"): Unit = {
-  Try(%("tar", "--exclude=provisioner", "-zcf", s"$sandboxFilePrefix.tar.gz", "sandboxes"))
+def zipLogs(logFileName: String = "ci.log", sandboxFile: String = "sandboxes"): Unit = {
```
Comment (alenkacz) [inline ci/pipeline]: sandboxFileName? :) since it's not the file itself

Comment (jeschkies) [reply ci/pipeline]: ✅

**Pull request merged**